lib/msf/core/exploit/sqli/mysqli/boolean_based_blind.rb

```diff
@@ -0,0 +1,52 @@
+#
+#   Boolean-Based Blind SQL injection support for MySQL
+#
+class Msf::Exploit::SQLi::MySQLi::BooleanBasedBlind < Msf::Exploit::SQLi::MySQLi::Common
+  #
+  #   Gets the output of the given SQL query, in a boolean-based blind manner.
+  #   The block given to initialize must return true if querying its parameter
+  #   gave a result, false otherwise.
+  #   @param query [String] The SQL query to execute
+  #   @param output_charset [Range] The range of characters to expect in the output, optional
+  #   @return [String] The query results
+  #
+  def run_sql(query, output_charset = nil)
```

Comment:
```suggestion
  def run_sql(query, output_charset: nil)
```

Prefer named optional arguments.

Also note that the `run_sql` has 2 signatures, can that be avoided?
```
def run_sql(query)
```

Reply:
For the different signatures, having output_charset in non-blind injections is useless, they are primarly there to save time and requests, also, run_sql is not the method users should use to dump data from a table, `dump_table_fields` and other high-level enumeration methods are there for that, it's a lower-level method to be used when we need to run custom queries, or we only care about one thing, and want to retrieve it fast (session_id for example in eyesofnetwork).

I could rename it, make it `blind_run_sql`, and make `run_sql` call it with output_charset being `nil`, if that has advantages, please let me know (for me, different signatures with optional additional options are okay).